### PR TITLE
feat(wren-ai-service): Add Question Rephrasing Pipeline for Question Recommendation

### PR DESCRIPTION
**Pull request by paopa**
## Summary
This PR introduces a new Question Rephrasing pipeline that improves question clarity and specificity while maintaining the original intent. The pipeline is integrated with the Question Recommendation service to enhance user query.

## Key Changes
1. Added new `QuestionRephrasing` pipeline class with the following features:
   - Takes user questions and database context as input
   - Uses LLM to rephrase questions for better clarity

2. Integration with Question Recommendation service:
   - Questions are now rephrased before SQL generation in the validation

3. Configuration updates:
   - Added `question_rephrasing` pipeline configuration across deployment files
   - Uses `gpt-4o-mini-2024-07-18` model

4. Added comprehensive test suite for the Question Rephrasing pipeline

## Configuration Changes
Added to the following config files:
- `deployment/kustomizations/base/cm.yaml`
- `docker/config.example.yaml`
- `tools/config/config.example.yaml`
- `tools/config/config.full.yaml`

## Summary by CodeRabbit

- **New Features**
	- Introduced a new pipeline for question rephrasing, enhancing the AI service's ability to handle user queries.
	- Added configuration entries for the question rephrasing pipeline in multiple configuration files.

- **Bug Fixes**
	- Improved question validation process by allowing dynamic configuration settings.

- **Tests**
	- Implemented asynchronous unit tests for the new question rephrasing functionality, covering various scenarios and edge cases.

### CODE REVIEW

**Comment by cyyeh** [comment]
@coderabbitai full review

File: wren-ai-service/src/pipelines/generation/question_rephrasing.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+3. Rephrase the question to be more clear and specific, using proper table and column names from the schema where relevant
+4. Ensure the rephrased question aligns with the actual data structure
+5. Maintain the original meaning and requirements
+6. Use natural, conversational language
+7. Keep the rephrased question concise
+8. Avoid referencing tables or columns that don't exist in the schema
+
+### OUTPUT FORMAT
+
+Return the rephrased question as a simple string that accurately reflects both the user's intent and the database structure.
```
Comment (cyyeh) [inline wren-ai-service/src/pipelines/generation/question_rephrasing.py:36]: I suggest still mentioning output format and telling llm to generate JSON here. Since not llms support structured output as of now

**Review by cyyeh — COMMENTED** [review]
overall LGTM

**Comment by paopa** [comment]
Based on the summary in https://www.notion.so/canner/7b0901e293004f058494a5a09872fe96?p=16692976d0bf8018aeb5fca1f798d8f8&pm=c. we close this PR, but the branch will be kept on the remote.
